doc/developer/design/20220322_multi_table_transactions.md

```diff
@@ -0,0 +1,48 @@
+# Multi-table Transactions
+
+## Summary
+
+Upcoming persist changes will remove the persist API allowing writing to multiple collections in an atomic operation.
+In order to allow the persist API to be written, we must do one of:
+
+1. Disallow multi-table transactions.
+2. Use an existing (or upcoming) materialize API to record transactions.
+3. Use a third-party product to record transactions.
+
+We need a log to record transactions that can write the entire transaction atomically, which will allow the rest of the system to know that the transaction committed.
+The controller writes transactions to the WAL, then if committed, each collection can apply their portion of the writes asynchronously.
```

Comment:
I think "if committed" here is a little confusing here for the same reasons mentioned above.